controllers/scopetemplate_controller_test.go

```diff
@@ -0,0 +1,373 @@
+		BeforeEach(func() {
+			scopeTemplate = &operatorsv1.ScopeTemplate{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "ScopeTemplate",
+					APIVersion: "operators.io.operator-framework/v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "scopetemplate-sample",
+				},
+				Spec: operatorsv1.ScopeTemplateSpec{
+					ClusterRoles: []operatorsv1.ClusterRoleTemplate{
+						{
+							GenerateName: "test",
+							Rules: []rbacv1.PolicyRule{
+								{
+									APIGroups: []string{
+										"",
+									},
+									Verbs: []string{
+										"get", "watch", "list",
+									},
+									Resources: []string{
+										"secrets",
+									},
+								},
+							},
+							Subjects: []rbacv1.Subject{
+								{
+									Kind:     "Group",
+									APIGroup: "rbac.authorization.k8s.io",
+									Name:     "manager",
+								},
+							},
+						},
+					},
+				},
+			}
+			Expect(k8sClient.Create(ctx, scopeTemplate)).NotTo(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			Expect(k8sClient.Delete(ctx, scopeTemplate)).NotTo(HaveOccurred())
+		})
+
+		It("should not create a clusterRole", func() {
+			labels := map[string]string{scopeTemplateUIDKey: string(scopeTemplate.GetUID()),
+				clusterRoleGenerateKey: "test"}
+
+			clusterRoleList := listCluserRole(0, labels)
+			Expect(clusterRoleList.Items).Should(BeNil())
+		})
+
+		When("a scopeInstance is created that references the scopeTemplate with a single namespace", func() {
+			BeforeEach(func() {
+				scopeInstance = &operatorsv1.ScopeInstance{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "ScopeInstance",
+						APIVersion: "operators.io.operator-framework/v1",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: scopeInstanceName,
+					},
+					Spec: operatorsv1.ScopeInstanceSpec{
+						ScopeTemplateName: scopeTemplate.GetName(),
+						Namespaces:        []string{namespace.GetName()},
+					},
+				}
+				err := k8sClient.Create(ctx, scopeInstance)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			AfterEach(func() {
+				Expect(k8sClient.Delete(ctx, scopeInstance)).NotTo(HaveOccurred())
+			})
+
+			It("should create the expected clusterRole", func() {
+				labels := map[string]string{scopeTemplateUIDKey: string(scopeTemplate.GetUID()),
+					clusterRoleGenerateKey: scopeTemplate.Spec.ClusterRoles[0].GenerateName}
+
+				clusterRoleList := listCluserRole(1, labels)
+
+				existingRole := clusterRoleList.Items[0]
+
+				Expect(len(existingRole.OwnerReferences)).Should(Equal(1))
+				Expect(existingRole.OwnerReferences).Should(ContainElement(metav1.OwnerReference{
+					APIVersion: "operators.io.operator-framework/v1",
+					Kind:       "ScopeTemplate",
+					Name:       scopeTemplate.GetName(),
+					UID:        scopeTemplate.GetUID(),
+				}))
+				Expect(existingRole.Rules).Should(Equal([]rbacv1.PolicyRule{
+					{
+						Verbs:     []string{"get", "watch", "list"},
+						APIGroups: []string{""},
+						Resources: []string{"secrets"},
+					},
+				}))
```

Comment:
Non-blocking: Let's create functions to check verify the clusterRole and clusterRoleBindings in a followup PR.

Reply:
added as TODO.